File: Difficulty: Easy/K-Pangrams/kpangrams.cpp
//{ Driver Code Starts
#include <bits/stdc++.h>
using namespace std;


// } Driver Code Ends
class Solution {
  public:

    bool kPangram(string str, int k) {
        unordered_set<char> st;
        int cntSps = 0;
        for(auto ch : str) 
        {    
            if(ch != ' ')
               st.insert(ch);
            else
               cntSps++;
        }
        
        if(26 - st.size() <= k && str.size()-cntSps >= 26) return true;
        
        return false;
    }
};

//{ Driver Code Starts.

int main() {
    int t;
    cin >> t;
    while (t--) {
        cin.ignore();
        string str;
        getline(cin, str);

        int k;
        cin >> k;

        Solution ob;
        bool a = ob.kPangram(str, k);
        if (a)
            cout << "true" << endl;
        else
            cout << "false" << endl;
    }
    return 0;
}
// } Driver Code Ends

File: Difficulty: Easy/K-Pangrams/kpangrams.java
//{ Driver Code Starts
// Initial Template for Java
import java.io.*;
import java.util.*;

class GFG {
    public static void main(String args[]) throws IOException {
        Scanner sc = new Scanner(System.in);
        int t = Integer.parseInt(sc.nextLine().trim());
        while (t-- > 0) {
            String str = sc.nextLine();
            int k = Integer.parseInt(sc.nextLine().trim());
            Solution obj = new Solution();
            if (obj.kPangram(str, k))
                System.out.println("true");
            else
                System.out.println("false");
        }
    }
}
// } Driver Code Ends


// User function Template for Java
class Solution {
    boolean kPangram(String str, int k) {
        // code here
                int count=0;
        if(str.replace(" ", "").length() < 26)
        return false;
        
        for(char i='a';i<='z';i++)
       {
          if(!str.contains(String.valueOf(i)))
              count++;  
       }
       if(count > k){
          return false;
       }
       return true;
    }
}